include/cuco/bit_vector.cuh

```diff
@@ -59,51 +91,100 @@ class bit_vector {
   bit_vector();
   ~bit_vector();
 
-  void add(bool bit);  // adds a new bit at the end
-  void build();        // builds indexes for rank and select.
-
+  /**
+   * @brief adds a new bit at the end
+   *
+   * @param bit Boolean value of new bit to be added
+   */
+  void add(bool bit);
+
+  /**
+   * @brief Builds indexes for rank and select
+   *
+   * Also creates device-side snapshot
+   */
+  void build();
+
+  /**
+   * @brief Modifies a single bit
+   *
+   * @param key position of bit to be modified
+   * @param bit new value of bit
+   */
   void set(Key key, bool bit);
```

Comment:
So since we build the trie on the host, we need this point-wise accessor available from the host side. Typically, cuco data structures reside on the device only. To access the data from the host, we provide bulk-operations, which operate on multiple input elements which are executed together in a kernel. I'd say we go with your initial bit vector implementation for now and put it in the `detail/trie` directory, i.e., not expose it publicly. That way, you can use your existing implementation and don't have to think about a generalized cuco-esque bit vector class. I'm going to put that on my todo list and once it's working we can think of switching to that implementation in the future.

Reply:
I moved bit_vector implementation into detail/trie. See commit 881d8335ce.

Regarding bulk insert operations, one distinction from other cuco structures is there is 'build' operation after all the keys are inserted. This involves building additional index structures and creating a device-side snapshot. Even if we maintain a device copy only (avoiding the snapshot), does it make sense to have 'build()' as part of the interface?